tests/Unit/Helpers/ParallelAlgorithms/NonlinearSolver/Algorithm.hpp

```diff
@@ -0,0 +1,232 @@
+  static type create_from_options(const type& option) { return option; }
+  static std::string name() noexcept { return "ExpectedResult"; }
+};
+
+// The vector `x` we want to solve for
+struct VectorTag : db::SimpleTag {
+  using type = DenseVector<double>;
+  static std::string name() noexcept { return "VectorTag"; }
+};
+
+using fields_tag = VectorTag;
+using nonlinear_source_tag = ::Tags::FixedSource<fields_tag>;
+
+// Checks for the correct solution after the algorithm has terminated.
+template <typename OptionsGroup>
+struct TestResult {
+  using const_global_cache_tags = tmpl::list<ExpectedResult>;
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ActionList, typename ParallelComponent>
+  static std::tuple<db::DataBox<DbTagsList>&&, bool> apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& cache,
+      const int /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    const auto& has_converged =
+        get<Convergence::Tags::HasConverged<OptionsGroup>>(box);
+    SPECTRE_PARALLEL_REQUIRE(has_converged);
+    SPECTRE_PARALLEL_REQUIRE(has_converged.reason() ==
+                             Convergence::Reason::AbsoluteResidual);
+    const auto& result = get<VectorTag>(box);
+    const auto& expected_result = get<ExpectedResult>(cache);
+    for (size_t i = 0; i < expected_result.size(); i++) {
+      SPECTRE_PARALLEL_REQUIRE(result[i] == approx(expected_result[i]));
+    }
+    return {std::move(box), true};
+  }
+};
+
+struct InitializeElement {
+  using const_global_cache_tags = tmpl::list<Source, InitialGuess>;
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ActionList, typename ParallelComponent>
+  static auto apply(db::DataBox<DbTagsList>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::GlobalCache<Metavariables>& cache,
+                    const int /*array_index*/, const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    const auto& b = get<Source>(cache);
+    const auto& x0 = get<InitialGuess>(cache);
+
+    return std::make_tuple(::Initialization::merge_into_databox<
+                           InitializeElement,
+                           db::AddSimpleTags<fields_tag, nonlinear_source_tag>>(
+        std::move(box), DenseVector<double>(x0), DenseVector<double>(b)));
+  }
+};
+
+template <typename Metavariables>
+struct ElementArray {
+  using chare_type = Parallel::Algorithms::Array;
+  using array_index = int;
+  using metavariables = Metavariables;
+
+  using nonlinear_solver = typename Metavariables::nonlinear_solver;
+  using linear_solver = typename Metavariables::linear_solver;
+
+  /// [action_list]
```

Comment:
Is this included somewhere as a snippet? I don't see it in the other files in this PR... is not, then remove the tag